bridge/setu/listener/base.go

```diff
@@ -169,10 +175,30 @@ func (bl *BaseListener) StartPolling(ctx context.Context, pollInterval time.Dura
 	for {
 		select {
 		case <-ticker.C:
-			header, err := bl.chainClient.HeaderByNumber(ctx, nil)
+			var bHeader *blockHeader
+
+			header, err := bl.chainClient.HeaderByNumber(ctx, number)
 			if err == nil && header != nil {
-				// send data to channel
-				bl.HeaderChannel <- header
+				if number != nil {
+					// finalized was requested
+					bHeader = &blockHeader{header: header, isFinalized: true}
+				} else {
+					// latest was requested
+					bHeader = &blockHeader{header: header, isFinalized: false}
+				}
+			}
+
+			// if error occured and finalized was requested, fall back to latest block
+			if err != nil && number != nil {
```

Comment:
Is the case `if err != nil && number == nil` possible? Should we cover it?

Reply:
Shall we log the error in case it's not nil?

Reply:
Well, in that case we don't need to send the data to the channel. 

I didn't cover it because it was the same earlier as well. See https://github.com/maticnetwork/heimdall/blob/develop/bridge/setu/listener/base.go#L172-L176.

Reply:
This is a case when the matic chain listener will call this function with `number` as `nil`, as we need the latest block and not the finalized block.

Reply:
Okay logging makes sense. Will do. Thanks!